api/base/serializers.py

```diff
@@ -177,8 +234,27 @@ def to_representation(self, obj, envelope='data'):
         meta = getattr(self, 'Meta', None)
         type_ = getattr(meta, 'type_', None)
         assert type_ is not None, 'Must define Meta.type_'
-        data = super(JSONAPISerializer, self).to_representation(obj)
-        data['type'] = type_
+
+        data = collections.OrderedDict([('id', ''), ('type', type_), ('attributes', collections.OrderedDict()),
+                                        ('relationships', collections.OrderedDict()), ('links', {})])
+
+        fields = [field for field in self.fields.values() if not field.write_only]
+
+        for field in fields:
+            try:
+                attribute = field.get_attribute(obj)
+            except SkipField:
+                continue
+
+            if isinstance(field, HyperlinkedIdentityFieldWithMeta):
```

Comment:
OK, I see.

However, shouldn't we be using a subclass of `HyperlinkedRelatedFields` rather than `HyperlinkedIdentityField`? It seems that we're trying to serialize *relationships* (e.g. a user's nodes) after all.